### PR TITLE
Fixes #76 - missing name and label in new config

### LINKED ISSUE #76: The configuration "metatag.instance.node.page" must have the "name" attribute specified.
Error message when trying sync configuration:

> The configuration "metatag.instance.node.page" must have the "name" attribute specified.

I don't see this option in the settings. The global metags have "name" attribute but not these instances. This prevents any configuration syncing. I may have to disable the module until this can be fixed.

### PR DESCRIPTION
**Pull request by yorkshire-pudding**
Fixes #76 

This adds in `name` and `label` attributes for any new metatag config instances.
It also has an update hook to fix any existing metatag configs.

### CODE REVIEW

File: metatag.module [hunk 1]
```diff
@@ -365,6 +365,12 @@ function metatag_config_save($data) {
   if ($config->get('storage') == METATAG_STORAGE_DEFAULT) {
     $config->set('storage', METATAG_STORAGE_OVERRIDDEN);
   }
+  if ($config->get('name') == NULL) {
+    $config->set('name',$name);
```
Comment (klonos) [inline metatag.module]:
```suggestion
    $config->set('name', $name);
```

File: metatag.module [hunk 2]
```diff
@@ -365,6 +365,12 @@ function metatag_config_save($data) {
...
+  }
+  if ($config->get('label') == NULL ) {
```
Comment (klonos) [inline metatag.module]:
```suggestion
  if ($config->get('label') == NULL) {
```

File: metatag.module [hunk 3]
```diff
@@ -2137,11 +2143,6 @@ function metatag_preprocess_page(&$variables) {
  * @see metatag_preprocess_page().
  */
 function metatag_preprocess_maintenance_page(&$variables) {
-  // Don't modify install or update pages.
-  if (defined('MAINTENANCE_MODE') && in_array(MAINTENANCE_MODE, array('install', 'update'))) {
-    return;
-  }
-
```
Comment (klonos) [inline metatag.module]: This change here seems to be unrelated to the problem we are trying to fix.

File: metatag.install [hunk 4]
```diff
@@ -544,3 +544,22 @@ function metatag_update_1004() {
   }
   config_set('metatag.settings', 'enabled_tags', $enabled_tags);
 }
+
+/**
+ * Add name and label attributes to any config instances that don't have one
+ */
+
+ function metatag_update_1005() {
```
Comment (klonos) [inline metatag.install]:
```suggestion
function metatag_update_1005() {
```

File: metatag.install [hunk 5]
```diff
@@ -544,3 +544,22 @@ function metatag_update_1004() {
...
+  $files = config_get_names_with_prefix('metatag.instance');
+  $configs = config_load_multiple($files);
+  foreach ($configs as $filename => $config) {
+    $instance = $config['instance'];
+    if (!isset($config['name']) ) {
```
Comment (klonos) [inline metatag.install]:
```suggestion
    if (!isset($config['name'])) {
```

File: metatag.install [hunk 6]
```diff
@@ -544,3 +544,22 @@ function metatag_update_1004() {
...
+      $name = str_replace(':', '.', $instance);
+      config_set($filename, 'name', $name);
+    }
+    if (!isset($config['label']) ) {
```
Comment (klonos) [inline metatag.install]:
```suggestion
    if (!isset($config['label'])) {
```

File: metatag.install [hunk 7]
```diff
@@ -544,3 +544,22 @@ function metatag_update_1004() {
...
+      config_set($filename, 'label', metatag_config_instance_label($instance));
+    }
+  }
+ }
```
Comment (klonos) [inline metatag.install]:
```suggestion
}
```

...the "no new line at end of file" issue will need to be fixed as well.

File: metatag.module (hunk 3 above)
Comment (yorkshire-pudding) [reply metatag.module]: No idea how I managed to delete this.

File: metatag.module (hunk 1 above)
Comment (yorkshire-pudding) [reply metatag.module]: Done

File: metatag.module (hunk 2 above)
Comment (yorkshire-pudding) [reply metatag.module]: Done

File: metatag.install (hunk 4 above)
Comment (yorkshire-pudding) [reply metatag.install]: Done

File: metatag.install (hunk 5 above)
Comment (yorkshire-pudding) [reply metatag.install]: Done

File: metatag.install (hunk 6 above)
Comment (yorkshire-pudding) [reply metatag.install]: Done

File: metatag.install (hunk 7 above)
Comment (yorkshire-pudding) [reply metatag.install]: Done

File: metatag.install [hunk 8]
```diff
@@ -544,3 +544,22 @@ function metatag_update_1004() {
   }
   config_set('metatag.settings', 'enabled_tags', $enabled_tags);
 }
+
+/**
+ * Add name and label attributes to any config instances that don't have one
+ */
+
+function metatag_update_1005() {
```
Comment (klonos) [inline metatag.install]: Extra line creeped in there, between the function and its docblock above it:

```suggestion
 */
function metatag_update_1005() {
```

Comment (yorkshire-pudding) [reply metatag.install]: Done
